### ISSUE TITLE
network services does not building VRF and iBGP peering in overlay if neighbor has services defined

### ISSUE THREAD

**Issue by philippebureau** [aristanetworks/ansible-avd#3804]
### Issue Summary

For mlag pairs,  if side A has services defined (ex: l3 interface) but side B does not.  AVD will not build VRF definition and iBGP peering in the overlay on side B node.  eos_designs should automatically build that to improve resiliency for single attached services.

Workaround: use node_type.filter.always_include_vrfs_in_tenants

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible AWX

### Steps to reproduce

```shell
-define a mlag pair in node_type
-define a tenant with a l3_interface on side A of the mlag pair.
-> switch A config will include iBGP peering with side B but Side B will not have the VRF or iBGP peering
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

**Comment by gmuloc** [comment]
cross-posting the closing message of #4530 

TL;DR: The penalty hit of 70 to 80% in the fact phase just to solve the corner case is not worth it. Use the workaround described in the issue.

So after careful consideration we have decided with @ClausHolbechArista to close this PR and deem it not feasible at the time with the existing code structure.

To be able to know which VRFs are in my MLAG peer I need to compute the filtered_tenants in the fact phase which is taking a lot of extra time. We have observed a 70-80% time penalty during the fact phase on the repo which is not something that we think is a good solution to impose on every AVD users for such a corner case.

I will remove the issue from AVD 5.0 milestone and we will take another stab at it later once the code structure has changed.
